private/bufpkg/bufimage/util.go

```diff
@@ -118,6 +206,26 @@ func imageWithOnlyPaths(image Image, fileOrDirPaths []string, allowNotExist bool
 				return nil, fmt.Errorf("path %q has no matching file in the image", potentialDirPath)
 			}
 		}
+		// Since all exclude paths are subset of the provided target paths, we can use the same
+		// check against the pruned exclude paths.
+		// TODO: similar to the comment above, should we be checking against all provided paths?
+		// It would require an expensive operation based on image.Files().
+		var pathsWithNoMatchingFiles []string
+		for excludeFileOrDirPath := range excludeFileOrDirPathMap {
+			var foundPath bool
+			for nonImportPath := range allNonImportPaths {
+				if normalpath.EqualsOrContainsPath(excludeFileOrDirPath, nonImportPath, normalpath.Relative) {
+					foundPath = true
+					break
+				}
+			}
+			if !foundPath {
+				pathsWithNoMatchingFiles = append(pathsWithNoMatchingFiles, excludeFileOrDirPath)
+			}
+		}
+		if len(pathsWithNoMatchingFiles) > 0 {
+			return nil, fmt.Errorf("paths with no matching files in the image: %v", pathsWithNoMatchingFiles)
+		}
```

Comment:
Could we reuse the `matchingPotentialDirPathMap` to validate the `--exclude` flags, too? We just need to verify that the paths specified by `--exclude` were matched, so I feel like we should be able to reuse it, right?

If so, this might simplify the case above, too.

Reply:
Ah yeah, basically during the `imageFile` ranging, we could also do a check for the excludes. Yeah, I think this actually works out well, since I'm reworking the logic for the file path "merging"/checking.